cmd/server/main.go

```diff
@@ -239,3 +235,63 @@ func downloadPluginFile(name, version string) (*os.File, int64, error) {
 
 	return tmpfile, written, nil
 }
+
+func lookupOrScanPlugin(plugin string, version string) (*shared.Scan, error) {
+	var s []byte
+	found := false
+
+	db.View(func(tx *bolt.Tx) error {
+		b := tx.Bucket([]byte(plugin))
+		if b == nil {
+			return nil
+		}
+
+		s = b.Get([]byte(version))
+		if s == nil {
+			return nil
+		}
+
+		found = true
+
+		return nil
+	})
+
+	if found {
+		var scan shared.Scan
+		err := json.Unmarshal(s, &scan)
```

Comment:
The marshalling/unmarshalling is probably a good candidate for an interface to the database. Having it mucking around in the logic around here is getting clumsy, since we're doing more with it now.